This is f2f09fb32946a806f46285419fe6359f30206811
"Hardcode the list of ELF targets here rather than using a glob.",
backported to 3.4. Notably, lld didn't support MIPS back in 3.4.

diff --git a/lld/lib/ReaderWriter/ELF/CMakeLists.txt b/lld/lib/ReaderWriter/ELF/CMakeLists.txt
index 39bb80c4470025..07d758002c404b 100644
--- a/lld/lib/ReaderWriter/ELF/CMakeLists.txt
+++ b/lld/lib/ReaderWriter/ELF/CMakeLists.txt
@@ -17,11 +17,7 @@ target_link_libraries(lldELF
 
 include_directories(.)
 
-file(GLOB all_valid_subdirs RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} "*/CMakeLists.txt")
-
-foreach(dir ${all_valid_subdirs})
-    if(${dir} MATCHES "^([^/]*)//CMakeLists.txt")
-        string(REGEX REPLACE "^([^/]*)//CMakeLists.txt" "\\1" dir_trimmed ${dir})
-        add_subdirectory(${dir_trimmed})
-    endif()
-endforeach(dir)
+add_subdirectory(Hexagon)
+add_subdirectory(PPC)
+add_subdirectory(X86)
+add_subdirectory(X86_64)
